### ISSUE TITLE
When using table keys, NVDA focus does not move to the second and third columns. Additionally, each cell in the first column is announced as collapsed.: A11y_AzureToolsStorageExplorer_GetSharedAccessSignature_ScreenReader

### ISSUE THREAD

**Issue by kapilvaishna** [microsoft/AzureStorageExplorer#8378]
### Preflight Checklist

- [x] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [x] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [x] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.36.2

### Architecture

x64

### Platform

Windows

### OS Version

Windows 11

## GitHub Tags:

#A11yMAS; #A11yTCS; #Win32; #DesktopApp; #A11ySev2; #BM_AzureToolsStorageExplorer_Win32_Dec2024; #Azure Tools Storage Explorer; #Screenreader; #WCAG1.3.1; #FTP; #NVDA; 

## Environment Details:
Application: Microsoft azure storage explorer
Version: 1.36.2
OS: Windows 11 Enterprise 24H2
Build: 26100.2605
NVDA Version: 2024.4.1
JAWS Version: 2024.2312.53

## Repro Steps 

1. Turn on NVDA
2. Launch the Azure Tools Storage Explorer.
3. TAB till "Blob container" in tree view and press ENTER key to expand it.
4. Create blob, delete any other blob, upload file under blob
5. TAB till activities tab and press ENTER key.
6. Navigate with down arrow key data present under activities tab
8. Observe that when using table keys, NVDA focus does not move to the second and third columns. Additionally, each cell in the first column is announced as collapsed.

## Actual Experience

Column two and three of the table are not accessible using the table keys (Ctrl + Alt + Arrow key). When using NVDA, pressing Ctrl + Alt + Right arrow key moves the focus directly from the first column to the fourth column, bypassing the second and third columns Additionally, each cell in the first column is announced as collapsed.
Note: Issue is not repro'ing JAWS.

## Expected Experience

Complete table should be accessible with table key. and as cells are not expandable and collapsible NVDA should not announce expand/collapsed.

## User Impact:
Screen reader user will get confused because focus will not move to second and third column and each cell in the first column is announced as collapsed.

## Attachment
![Image](https://github.com/user-attachments/assets/3e64c981-01c1-4cf6-b1eb-9de4ab01d640)

JAWS:
![Image](https://github.com/user-attachments/assets/bc46df8e-d920-49b9-8070-152d6deecea8)

Narrator:
![Image](https://github.com/user-attachments/assets/576d1087-3be7-4a68-b6e7-eb2bb9121b42)

**Comment by craxal** [comment]
@kapilvaishna

* The Activities tab is indeed implemented as a table, and this is by design. There are multiple columns in the table, including status, message text, speed, progress, and actions. When there are multiple activities running, the table appearance is easier to see.
* The "collapsed" announcement is also by design. The Activities tab is actually an implementation of a tree grid. Although it happens rarely, some activities spawn child activities. These activities are nested under their parents in the Activities tab. As such, to our knowledge, we are following appropriate guidelines by announcing that items are collapsed. However, it would make sense that, if an item does not have child items, we should not have a collapsed/expanded state at all. Should this be changed?

**Comment by kapilvaishna** [comment]
@craxal  Please find screen reader behavior below which make issue for screen reader user. please provide your observation and should i update this bug as per this issue close this issue or log a new bug. 

1. Column two and three of the table are not accessible using the table keys (Ctrl + Alt + Arrow key). When using NVDA, pressing Ctrl + Alt + Right arrow key moves the focus directly from the first column to the fourth column, bypassing the second and third columns. As a result, the table keys do not move the focus to columns two and three. Please find Gif in bug description.
2. Yes, NVDA announced collapsed/expand this should be fix.

**Comment by craxal** [comment]
@kapilvaishna 
1. This behavior is expected, because there is nothing in those columns. They contain speed and progress information, which not every activity provides, and we only display when the activity status is in progress. However, we can modify this from an accessibility perspective if it's too confusing.
2. Thank you, we'll look into it.

**Comment by kapilvaishna** [comment]
@craxal 
1. Thank you updated the bug as per issue.

**Comment by craxal** [comment]
@kapilvaishna I can add table headers. However, on closer inspection of the code, I observe that we are setting the `aria-colspan` property on the message column. So if the speed or progress columns are blank, the message column clearly indicates that it spreads across multiple columns. If the screen reader doesn't use the `aria-colspan` property, well, that's their problem, not ours.
 
Can you verify if this reproduces with VoiceOver in Mac? Can you also check if the aria-colspan is what you would expect (by using Accessibility Insights for example)?

**Comment by craxal** [comment]
@kapilvaishna Fix is ready for validation.

**Comment by kapilvaishna** [comment]
@craxal From my understanding, the "command to copy" button should be in the actions column, but it is currently under the speed column. Could you please confirm on this.
Additionally, is there a way for me to obtain data in all four columns? 

![Image](https://github.com/user-attachments/assets/4a3adb5d-d628-401c-912f-c2a850c5accd)

**Comment by craxal** [comment]
@kapilvaishna Sorry, I forgot to send you a link to the new build. The build number you should have should be at least 20250128.2 or later. In the future, can you please report the build number that you test on?

**Comment by craxal** [comment]
@kapilvaishna Have you been able to verify whether this is fixed?

**Comment by craxal** [comment]
@kapilvaishna The HTML is properly formed:

![Image](https://github.com/user-attachments/assets/fcfb9b35-9f6b-4748-9f09-925903194d49)

Even if I also add the `colspan` attribute, NVDA fails to read the correct column. There appears to be an existing issue with NVDA: https://github.com/nvaccess/nvda/issues/12482.

I didn't have problems with VoiceOver. Can you please verify whether VoiceOver on macOS reads the correct column?

**Comment by craxal** [comment]
I tried it with JAWS, too, and I also had a problem with it. Seems like the colspan is too much of a hassle, so maybe we'll remove that part.

**Comment by kapilvaishna** [comment]
Issue is not repro for version 1.38.0-dev(104) so closing the bug

![Image](https://github.com/user-attachments/assets/b53cf2b0-1946-4db6-84f4-ded5ce2d6964)

![Image](https://github.com/user-attachments/assets/3ba67c1c-29ec-436d-8dfd-27f673ed36c7)
